tools/simulator/main.cc

```diff
@@ -1,24 +1,33 @@
+#include <filesystem>
+
 #include <CLI/App.hpp>
 #include <CLI/Config.hpp>
 #include <CLI/Formatter.hpp>
 
 #include <spdlog/spdlog.h>
 
+#include "hart/hart.hh"
+
+namespace fs = std::filesystem;
+
 int main(int argc, char **argv) {
   CLI::App app{"Simulator"};
 
   bool isInfo;
   app.add_flag("-i,--info", isInfo, "Turn on info logging level");
 
+  // input
+  fs::path input{};                                    // input
+  app.add_option("input", input, "input")->required(); // input
```

Comment:
I suppose you should provide more descriptive comments (or remove them) and more informative help string to option